src/HexagonOptimize.cpp

```diff
@@ -1777,11 +1777,12 @@ class OptimizeShuffles : public IRMutator {
                 Expr index_span = span_of_bounds(index_bounds);
                 index_span = common_subexpression_elimination(index_span);
                 index_span = simplify(index_span);
-
-                if (can_prove(index_span < 256)) {
```

Comment:
What do you think would be a good number here? I don't think we need not restrict dynamic_shuffle to 256.

Reply:
I think we should keep it 256 until we know a better value for it. There must be some crossover point at which it is not more efficient to use dynamic_shuffle, but I don't know what it is.